### DISCUSSION TITLE
How do I replace users' avatars?

### DISCUSSION THREAD

**Discussion by this-xkit** [wg-easy/wg-easy#1359] [Q&A] [answered]
Gravatar support might be useful for some, but unfortunately, not for me. I tried to dive into the code to figure out how to replace the default grey avatars with something custom, but my lack of HTML knowledge didn’t allow me to sort it out on my own. I’m not talking about enabling the upload of custom avatars (which would be ideal, of course), but at least replacing the standard icons with something more interesting. Could anyone point me in the right direction on how to do this? I’d really appreciate it!

**Comment by kaaax0815** [comment, marked as answer]
There is currently no way to do that. But its noted in https://github.com/wg-easy/wg-easy/discussions/1179#discussioncomment-10264297
Its in the source code here
https://github.com/wg-easy/wg-easy/blob/66c64af7cf1d0d413b2d665d01e063d4f042cef4/src/www/index.html#L163-L167

**Reply by this-xkit** [reply]
Thx!

**Reply by this-xkit** [reply]
Maybe you could help me again? I added my avatar in SVG format, but it’s not displaying. PNG works fine, though. If I navigate to https://vpn.xkit.one/assets/avatar.svg, I can see that the image is available at that path.

```
<!-- Avatar -->
<div class="h-10 w-10 mt-2 self-start rounded-full bg-gray-50 relative">
  <img v-if="client.avatar" :src="client.avatar" class="w-10 rounded-full absolute top-0 left-0" />
  <img v-else src="/assets/avatar.svg" class="w-6 m-2 text-gray-300" />

  <div
    v-if="client.latestHandshakeAt && ((new Date() - new Date(client.latestHandshakeAt) < 1000 * 60 * 10))">
    <div
      class="animate-ping w-4 h-4 p-1 bg-red-100 dark:bg-red-100 rounded-full absolute -bottom-1 -right-1">
    </div>
    <div class="w-2 h-2 bg-red-800 dark:bg-red-600 rounded-full absolute bottom-0 right-0"></div>
  </div>
</div>
```

![image](https://github.com/user-attachments/assets/22f5efed-d6f6-4169-a3a7-0da9f9a1bcae)

**Reply by kaaax0815** [reply]
I don't know what's wrong as it looks fine. Maybe look in the Developer Tools for any errors

**Reply by this-xkit** [reply]
There are no errors in Developer Tools(

![image](https://github.com/user-attachments/assets/76cea5ad-7755-4038-97ca-472ce1e0f983)

**Reply by kaaax0815** [reply]
You have a filter in your console + check the network tab

**Reply by this-xkit** [reply]
I was unable to solve this problem. And if I upload a png, then the picture loads, but moves to the lower right corner, but since I am not an html developer, I did not find a quick solution. If I find it later, I'll post it here

**Reply by this-xkit** [reply]
It was possible to replace the avatar by replacing the svg code, if you upload an image file from the server, then all sorts of errors occur. A good example is below:

```
<div class="h-10 w-10 mt-2 self-start rounded-full bg-gray-50 relative">
<svg class="w-6 m-2 text-gray-300" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
<g id="user" fill="#396ce8">
<circle cx="12.009" cy="6.5" r="4" fill="#396ce8" style="fill: rgb(153, 27, 27);"></circle>
<path d="m14 12.5h-4c-4.06 0-5.5 2.973-5.5 5.519 0 2.281 1.211 3.481 3.5 3.481h8c2.292 0 3.5-1.2 3.5-3.481 0-2.546-1.44-5.519-5.5-5.519z" opacity=".4" fill="#396ce8" style="fill: rgb(153, 27, 27);"></path>
</g>
</svg>

<img v-if="client.avatar" :src="client.avatar" class="w-10 rounded-full absolute top-0 left-0" />
```

![image](https://github.com/user-attachments/assets/ef9ceca1-e7a8-45ec-9553-6b3620f5743c)

Now I also tried to change the color of the online client indicator from red to green. If you just change bg-red to bg-green, then the indicator is not displayed at all... I understood that you need to explicitly specify colors in css, but I haven't figured out where and how yet


```
<div
 v-if="client.latestHandshakeAt && ((new Date() - new Date(client.latestHandshakeAt) < 1000 * 60 * 10))">
<div
class="animate-ping w-4 h-4 p-1 bg-red-100 dark:bg-red-100 rounded-full absolute -bottom-1 -right-1">
</div>
<div class="w-2 h-2 bg-red-800 dark:bg-red-600 rounded-full absolute bottom-0 right-0"></div>
</div>
</div>
```

UPD: The color of the online client indicator also managed to be changed, it turned out that it was necessary to define colors in app.css. Now it seems to look more interesting :)

![image](https://github.com/user-attachments/assets/cd1d52c4-85fd-499e-9a38-a6dc58e327d4)

**Reply by kaaax0815** [reply]
The first issue I don't know the reason.

The second is because the tailwind classes are only generated if they are in the source code. If you use a different class you would have to regenerate the tailwind file
